Marble Blast Platinum/platinum/client/ui/mp/MPServerDlg.gui

```diff
@@ -520,6 +520,13 @@ function MPServerDlg::saveFields(%this) {
 	startHeartbeat();
 
 	PlayMissionGui.updateMPButtons();
+	MPServerDlg.schedule(50, updateServerPrivacy);
+}
+
+//This needs to be a seperate function because checking the password needs to be delayed for some reason. Thanks PQ
+function MPServerDlg::updateServerPrivacy(%this) {
+	commandToAll('UpdateServerPrivacy', ($MPPref::Server::Password !$= ""));
+	commandToAll('UpdateRichPresencePMG');
 }
```

Comment:
This should be done server-side rather than client side. I'm guessing the best place to start would be `platinum/server/scripts/settings.cs` in either `serverCmdServerSetting` or `serverSetSetting`. A client should only respect an UpdateServerPrivacy request coming from the server, not another client. That might possibly remove the need for a delay as well